modules/imgproc/src/hal_replacement.hpp

```diff
@@ -880,6 +880,25 @@ inline int hal_ni_gaussianBlur(const uchar* src_data, size_t src_step, uchar* ds
 #define cv_hal_gaussianBlur hal_ni_gaussianBlur
 //! @endcond
 
+/**
+   @brief Blurs an image using a symmetric Gaussian filter with square kernel and sigma=0.
+   @param src_data Source image data
+   @param src_step Source image step
+   @param dst_data Destination image data
+   @param dst_step Destination image step
+   @param width Source image width
+   @param height Source image height
+   @param depth Depth of source and destination image
+   @param cn Number of channels
+   @param ksize Width of kernel
+   @param border_type Border type
+*/
+inline int hal_ni_gaussianBlurSymmetric(const uchar* src_data, size_t src_step, uchar* dst_data, size_t dst_step, int width, int height, int depth, int cn, size_t ksize, int border_type) { return CV_HAL_ERROR_NOT_IMPLEMENTED; }
```

Comment:
1. ksize: `size_t => int` to match `width` and `height` types
2. add `int roi_x, int roi_y, int roi_width, int roi_height` to support ROI and external multi-threading
3. add `const uchar* border_value`

Reply:
classic gaussianBlur uses size_t for kernel size too:
```
inline int hal_ni_gaussianBlur(const uchar* src_data, size_t src_step, uchar* dst_data, size_t dst_step, int width, int height, int depth, int cn, size_t margin_left, size_t margin_top, size_t margin_right, size_t margin_bottom, size_t ksize_width, size_t ksize_height, double sigmaX, double sigmaY, int border_type) { return CV_HAL_ERROR_NOT_IMPLEMENTED; }
```

Reply:
border_value = scalarToRawData()

Reply:
"add const uchar* border_value" - I'll do with dedicated PR. Need to change public API first.